src/bios/flib/src/idt/idt.rs

```diff
@@ -0,0 +1,229 @@
+use alloc::vec::Vec;
+use core::arch::asm;
+use core::ptr::write_volatile;
+
+use modular_bitfield;
+use modular_bitfield::bitfield;
+use modular_bitfield::BitfieldSpecifier;
+use modular_bitfield::prelude::{B1, B11, B13, B2, B4, B8};
+
+use crate::debug;
+use crate::debug::debug::print_str;
+use crate::gdt::gdt::SegmentDescriptor;
+
+use numtoa::NumToA;
+use crate::interrupts::{disable_interrupts, enable_interrupts};
+
+pub enum GateType {
+    TaskGate = 0b0101,
+    InterruptGate16b = 0b0110,
+    InterruptGate32b = 0b1110,
+    TrapGate16b = 0b0111,
+    /// Most of the time you will chose a [GateType::TrapGate32b]
+    TrapGate32b = 0b1111,
+}
+
+/// [Table] contains entries that describes interrupts
+/// It has the following structure :
+///  ----------------------
+/// |   Address   |  Entry  |
+/// | --------------------- |
+/// |  Offset + 0 | entry 0 |
+/// |  Offset + 8 | entry 1 |
+///  -----------------------
+pub struct Table {
+    entries: Vec<GateDescriptor>,
+}
+
+impl Table {
+    /// Writes the [Table] to a given offset. You have to ensure that there is enough
```

Comment:
```suggestion
    /// Writes the `Table` to a given offset. You have to ensure that there is enough
```
Je suis d'avis d'éviter les liens vers elle-même quand on commente des élements de la struct